### PR TITLE
Hunspell freeling conflict

### PR DESCRIPTION
**Pull request by Moisan**
Following #42766, add conflicts_with between `hunspell` and `freeling`. I also added a test to `hunspell`.

### CODE REVIEW

File: Library/Formula/hunspell.rb [hunk 1]
```diff
@@ -35,7 +44,7 @@ def caveats; <<-EOS.undent
 +++ src/tools/hunspell.cxx	2013-08-02 18:20:27.000000000 +0200
 @@ -28,7 +28,7 @@
  #ifdef WIN32
-
```
Comment (DomT4) [inline Library/Formula/hunspell.rb]: Please try not to tweak the whitespacing in patches; doing so has the tendency to pass the CI and then break for users.

File: Library/Formula/hunspell.rb [hunk 2]
```diff
@@ -18,6 +20,8 @@ def install
     system "make"
     ENV.deparallelize
     system "make", "install"
+
+    pkgshare.install %w[tests]
```
Comment (DomT4) [inline Library/Formula/hunspell.rb]: Can just do `pkgshare.install "tests"`. No need for an array unless there's multiple things to install.

**Comment by bfontaine** [comment]
FYI we’ll be able to close #42766 when this will be merged.

**Comment by DomT4** [comment]
You may need to rebase this against the master branch here, I'm getting yelled at by git:

```
fatal: sha1 information is lacking or useless (Library/Formula/freeling.rb).
error: could not build fake ancestor
```

**Comment by bfontaine** [comment]
Thank you for your contribution to Homebrew @Moisan; we appreciate it! :tada:

**Pull request closed without merging**